### PR TITLE
Build against libnl from conda-forge

### PR DESCRIPTION
**Pull request by manopapad**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

Apparently the system `libnl-3.so` on CentOS has different symbols than on Ubuntu (the former uses `nla_get_u8@libnl_3`, whereas the latter uses `nla_get_u8`). This makes it impossible to build against the conda-forge `libibverbs.so` on Ubuntu, because it's referring to unavailable symbols.

Attempting to fix this by building against the `libnl` from `conda-forge`.

### CODE REVIEW

**Comment by manopapad** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
Thanks Manolis! 🙏

Could you please add...

```YAML
azure:
  upload_packages: False
```

...to...

https://github.com/conda-forge/rdma-core-feedstock/blob/e219de33f7a45e6303eb433d7cfa5617c34a5df4/conda-forge.yml#L1

...and then re-render?

This will cache the packages on CI so we can test

File: conda-forge.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+azure:
+  upload_packages: False
```
Comment (jakirkham) [inline conda-forge.yml:2]: Sorry copy-pasted these lines. This should be `True`

```suggestion
  upload_packages: True
```

**Comment by manopapad** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
Looking at the CI errors, think we need to move `libnl` to `requirements/host`. So it would look roughly like this

```yaml
requirements:
  build:
    ...
  host:
    - libnl
```

Currently `host` isn't a section we have here. So this would be adding a new section to `requirements`

**Comment by manopapad** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,14 +21,16 @@ requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
-    - {{ cdt("libnl3-devel") }}
+    - libnl
```
Comment (jakirkham) [inline recipe/meta.yaml:24]: Think we can drop this from `build` now that it is in `host`

CDTs are a bit special as they get added to `build`, but most cases dependencies that are linked against are added to `host`

```suggestion
```

**Comment by manopapad** [comment]
@conda-forge-admin, please rerender

File: conda-forge.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+azure:
+  upload_packages: true
```
Comment (jakirkham) [inline conda-forge.yml:2]:
```suggestion
  store_build_artifacts: true
```

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
Looks like the [CI artifacts]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=844356&view=artifacts&pathAsName=false&type=publishedArtifacts ) are available

Here are [some relevant docs on using CI artifacts locally]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#downloading-prebuilt-packages-from-ci )

Also here are examples with screenshots in issue: https://github.com/conda-forge/conda-forge.github.io/issues/1424

**Review by pentschev — APPROVED** [review]
Looks good to me, once @manopapad confirms working I think we can merge it. Thanks @manopapad ! 😄

**Comment by manopapad** [comment]
CI packages seem to work for me! One thing that's not clear to me how to check; will the final package contain a dependency on the conda-forge `libnl` package?

BTW I figured out why the Ubuntu (and actually also CentOS) `libnl.so` doesn't use the versioned symbols, resulting in the original problem. The upstream `libnl` linker script actually adds versioning to the symbols https://github.com/thom311/libnl/blob/main/libnl-3.sym, and the conda-forge package build respects this. However, Debian (and I assume other distros) actually patches-out this versioning https://sources.debian.org/patches/libnl3/3.7.0-0.2/debian/no-symvers.diff/, so they end up with a `libnl-3.so` with symbols that don't contain the version number.

**Comment by jakirkham** [comment]
That's good to hear!. Thanks Manolis 🙏

Yes we can see that at the end of the tests

```
    libnl:         3.8.0-hd590300_0  conda-forge
```

Am noticing we still are `yum install`ing the system package though. We also just updated `libnl`. So am going to push a change to this PR to fix the `yum` issue and start a fresh build with the new `libnl`

If things still look good after that change, we should be good to go

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
Looks like new [CI artifacts]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=845066&view=artifacts&pathAsName=false&type=publishedArtifacts ) are available

Could we try these as well to confirm things still work?

If they do, think we can merge

**Comment by manopapad** [comment]
The latest CI packages work for me!

File: conda-forge.yml [hunk 4]
```diff
@@ -1,3 +1,5 @@
+azure:
+  store_build_artifacts: true
```
Comment (jakirkham) [inline conda-forge.yml:2]: Since we are done debugging the packages, we can turn this off. Can always it back when needed

```suggestion
```

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Pull request merged**